srcCpp/Infrastructure_pro.cxx

```diff
@@ -539,8 +539,8 @@ bool configureShmemTransport(
     int retcode;
```

Comment:
Refered to line 511:

```
    DDS_Property_t *parentProp =
            DDSPropertyQosPolicyHelper::lookup_property(qos.property,
                    "dds.transport.shmem.builtin.parent.message_size_max");

    int parentMsgSizeMax = DEFAULT_MESSAGE_SIZE_MAX;
    if (parentProp != NULL && parentProp->value != NULL) {
        parentMsgSizeMax = atoi(parentProp->value);
    }
```

I believe, since you modified the place where this function is called, that the transport.messageSizeMax value is already populated. I believe we should use that one, instead of the message_size_max for just SHMEM, it will simplify code, plus, if you are setting SHMEM + UDPv4, the calculations will be the right ones.